test_server.py

```diff
@@ -223,6 +223,18 @@ def test_create_pet_with_no_related_product_id_and_type(self):
         resp = self.app.post('/recommendations', data=data, content_type='application/json')
         self.assertEqual( resp.status_code, HTTP_400_BAD_REQUEST )
 
+    def test_update_recommendation(self):
+        new_recommendation = {'parent_product_id': '2', 'priority': '5', 'related_product_id':'4', 'type': 'x-sell'}
+        data = json.dumps(new_recommendation)
+        resp = self.app.put('/recommendations/3', data=data, content_type='application/json')
+        self.assertEqual( resp.status_code, HTTP_200_OK )
+        new_json = json.loads(resp.data)
+        self.assertEqual (new_json['parent_product_id'], 2)
+
```

Comment:
I think an additional test is needed for a failed put which doesn't update
new_recommendation = {'parent_product_id': '3', 'priority': '5', 'related_product_id':'4', 'type': 'x-sell'}
put to '/recommendations/3'
self.assertEqual (new_json['type'], 'up-sell')